runtime/service_integ_test.go

```diff
@@ -351,71 +355,104 @@ func TestMultipleVMs_Isolated(t *testing.T) {
 				TimeoutSeconds: 300,
 			}
 
-			resp, err := fcClient.CreateVM(ctx, req)
-			require.NoError(t, err, "failed to create vm")
+			resp, createVMErr := fcClient.CreateVM(ctx, req)
+			if createVMErr != nil {
+				matches, err := findProcess(ctx, findFirecracker)
+				if err != nil {
+					return err
+				}
+				return fmt.Errorf(
+					"failed to create a VM while there are %d Firecracker processes: %w",
+					len(matches),
+					createVMErr,
+				)
+			}
 
-			var containerWg sync.WaitGroup
+			eg := &errgroup.Group{}
 			for containerID := 0; containerID < int(containerCount); containerID++ {
-				containerWg.Add(1)
-				go func(containerID int) {
-					defer containerWg.Done()
-					testMultipleExecs(
+				containerID := containerID
+				eg.Go(func() error {
+					return testMultipleExecs(
 						ctx,
-						t,
 						vmID,
 						containerID,
 						client, image,
 						jailerConfig,
 						resp.CgroupPath,
 					)
-				}(containerID)
+				})
 			}
 
 			// verify duplicate CreateVM call fails with right error
 			_, err = fcClient.CreateVM(ctx, &proto.CreateVMRequest{VMID: strconv.Itoa(vmID)})
-			require.Error(t, err, "did not receive expected error for duplicate CreateVM call")
+			if err == nil {
+				return fmt.Errorf("VM ID must be unique (err=%s)", err)
```

Comment:
If we return here and there are still goroutines running `testMultipleExecs`, those will keep running right? Would it be better to create the error group with a derived, cancellable context and `defer cancel()`?

Reply:
Right. Let me add context to the groups.